iceoryx_posh/test/moduletests/test_mepoo_config.cpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2021 by  Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/iceoryx_posh_types.hpp"
+#include "iceoryx_posh/mepoo/mepoo_config.hpp"
+#include "iceoryx_utils/internal/posix_wrapper/shared_memory_object/allocator.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+
+using namespace iox::mepoo;
+
+class MePooConfig_Test : public Test
+{
+  public:
+    void SetUp() override
+    {
+    }
+    void TearDown() override
+    {
+    }
+};
+
+TEST_F(MePooConfig_Test, AddMemPoolMethodAddsTheCorrespondingMempoolInTheMemPoolConfigContainer)
+{
+    MePooConfig sut;
+    constexpr uint32_t SIZE{128U};
+    constexpr uint32_t CHUNK_COUNT{100U};
+
+    sut.addMemPool({SIZE, CHUNK_COUNT});
+
+    EXPECT_EQ(sut.m_mempoolConfig.size(), 1U);
+    EXPECT_EQ(sut.m_mempoolConfig[0].m_size, SIZE);
+    EXPECT_EQ(sut.m_mempoolConfig[0].m_chunkCount, CHUNK_COUNT);
+}
+
+TEST_F(MePooConfig_Test, AddingMempoolWhenTheMemPoolConfigContainerIsFullReturnsError)
+{
+    MePooConfig sut;
+    constexpr uint32_t SIZE{128U};
+    constexpr uint32_t CHUNK_COUNT{100U};
+
+    for (size_t i = 0; i < iox::MAX_NUMBER_OF_MEMPOOLS; i++)
+    {
+        sut.addMemPool({SIZE, CHUNK_COUNT});
+    }
+    EXPECT_DEATH({ sut.addMemPool({SIZE, CHUNK_COUNT}); }, ".*");
```

Comment:
The test here is ok but the current implementation just throws a logmessage and terminates when it tries to add more mempools to the list than possible.
Personally i think that it is not fatal because RouDi is still operational with the reduced set of mempools. It make sense here to call the errorhandler with errorlevel SEVERE here because applications may be compromised or the functionality is reduced.
@elfenpiff FYI